akka-persistence-testkit/src/main/scala/akka/persistence/testkit/javadsl/UnpersistentBehavior.scala

```diff
@@ -0,0 +1,132 @@
+  }
+
+  def fromDurableState[Command, State](behavior: Behavior[Command]): UnpersistentBehavior[Command, Void, State] =
+    fromDurableState(behavior, null.asInstanceOf[State])
+
+  private val noEventProbe: PersistenceProbe[Void] =
+    new PersistenceProbe[Void] {
+      def getAllEffects(): List[PersistenceEffect[Void]] =
+        // could return an empty list, but the intent is that any use of this probe should fail the test
+        boom()
+
+      def extract(): PersistenceEffect[Void] = boom()
+      def expectPersistedClass[S <: Void](clazz: Class[S]): PersistenceEffect[S] = boom()
+      def hasEffects: Boolean = boom()
+      def expectPersisted(obj: Void): PersistenceProbe[Void] = boom()
+      def expectPersisted(obj: Void, tag: String): PersistenceProbe[Void] = boom()
+      def expectPersisted(obj: Void, tags: Set[String]): PersistenceProbe[Void] = boom()
+
+      private def boom() = throw new AssertionError("No events were persisted")
+    }
+}
+
+class UnpersistentBehavior[Command, Event, State] private (
+    behavior: Behavior[Command],
+    eventProbe: PersistenceProbe[Event],
+    stateProbe: PersistenceProbe[State]) {
+  def getBehavior(): Behavior[Command] = behavior
+
+  /** NB: durable state behaviors will not publish events to this probe */
+  def getEventProbe(): PersistenceProbe[Event] = eventProbe
+
+  def getStateProbe(): PersistenceProbe[State] = stateProbe
+  def getSnapshotProbe(): PersistenceProbe[State] = stateProbe
+}
+
+case class PersistenceEffect[T](persistedObject: T, sequenceNr: Long, tags: Set[String])
+
+trait PersistenceProbe[T] {
+
+  /** Collect all persistence effects from the probe and empty the probe */
+  def getAllEffects(): List[PersistenceEffect[T]]
+
+  /** Get and remove the oldest persistence effect from the probe */
+  def extract(): PersistenceEffect[T]
+
+  /** Get and remove the oldest persistence effect from the probe, failing if the
+   *  persisted object is not of the requested type
+   */
+  def expectPersistedClass[S <: T](clazz: Class[S]): PersistenceEffect[S]
+
+  /** Are there any persistence effects */
+  def hasEffects: Boolean
+
+  /** Assert that the given object was persisted in the oldest persistence effect and
+   *  remove that persistence effect
+   */
+  def expectPersisted(obj: T): PersistenceProbe[T]
+
+  /** Assert that the given object was persisted with the given tag in the oldest persistence
+   *  effect and remove that persistence effect.  If the persistence effect has multiple tags,
+   *  only one of them has to match in order for the assertion to succeed.
+   */
+  def expectPersisted(obj: T, tag: String): PersistenceProbe[T]
+
+  /** Assert that the given object was persisted with the given tag in the oldest persistence
+   *  effect and remove that persistence effect.  If the persistence effect has tags which are
+   *  not given, the assertion fails.
+   */
+  def expectPersisted(obj: T, tags: Set[String]): PersistenceProbe[T]
+}
```

Comment:
The various `expectNnnn` feel more natural, we did some preparation and now we expect there to be some effects based on that. 👍